README.md

```diff
@@ -1,30 +1,55 @@
-# Cookies in Sinatra
+# Cookies and Sessions
 
-### What exactly are cookies?
+## Objectives
 
-Cookies can be a confusing concept to grasp at first. They are simply tiny text files with a unique ID tag, and are stored on your computer's browser directory or application data subfolders. An identical cookie/text file is created in the company's database. A cookie keeps track of your movements within a web application (or website), and helps that web application remember pertinent data about you. For example, it helps them where you have left off, your registered login information, preferences, the last time you signed in, and so on.
+1. Define cookies. 
+2. Define sessions. 
+3. Learn the difference between sessions and cookies.
+
+## How do your websites recognize you?
+
+When you visit Facebook, chances are, you'll find you're logged in and be taken to your timeline directly. If you check your gmail account, you might see advertisements relating to search terms you typed into Google earlier that day. When you shop on Amazon, your shopping cart stays full while you continue to shop all the way up until you checkout. How is this possible? Is your computer spying on you all the time? The answer is, sort of. 
+
+Before we can understand what is so amazing about the kind of persistence and recognition we've come to expect from our web applications, we need to understand something about how the web works. 
+
+> HTTP is a stateless protocol, treating each request as an independent transaction that is unable to use information from any previous requests. This means there is no way within the hypertext transfer protocol to remember a user’s identity from page to page; instead, web applications requiring user login must use a session, which is a semi-permanent connection between two computers (such as a client computer running a web browser and a server running Rails).
+
+>–– Michael Hartl, *The Ruby on Rails Tutorial*
+
+A **session** is simply an object, like a hash, that stores data describing a client's interaction with a website at a given point in time. The session hash lives on the server. Your application can access it in any of your controllers at any point in time. 
+
+How does a session know who is interacting with a website? Through **cookies**. Cookies are are snippets of text placed on the user’s browser. Cookies persist from one page to the next when you are browsing the web. So, they can store information (such as a user id) that can then be used by the application to retrieve the logged-in user from the database, among other things. 
```

Comment:
doesn't a server know who is interacting with a website, by using cookies and session data